File: Euclid/Messages/Incoming/INITUNITLISTENER.cs
using Euclid.Game;
using Euclid.Messages.Outgoing;
using Euclid.Network.Streams;
using Euclid.Storage.Database.Access;

namespace Euclid.Messages.Incoming
{
    class INITUNITLISTENER : IMessageEvent
    {
        public void Handle(Player player, Request request)
        {
            player.Send(new ALLUNITS(RoomManager.Instance.ReplaceQueryRooms(RoomDao.GetPublicFlats())));
        }
    }
}

File: euclid/src/main/java/euclid/messages/incoming/INITUNITLISTENER.java
package euclid.messages.incoming;

import euclid.game.Player;
import euclid.messages.IMessageEvent;
import euclid.messages.outgoing.ALLUNITS;
import euclid.network.streams.Request;
import euclid.storage.database.access.RoomDao;
import euclid.game.room.RoomManager;

public class INITUNITLISTENER implements IMessageEvent {

    @Override
    public void handle(Player player, Request request) {
        player.send(new ALLUNITS(RoomManager.getInstance().replaceQueryRooms(RoomDao.getPublicFlats())));
    }
}
